packages/react-chat/src/views/ChatWindow/ChatWindowStandaloneView.tsx

```diff
@@ -0,0 +1,61 @@
+import { useContext, useEffect } from 'react';
+
+import { isObject } from '@/common/utils';
+import { RuntimeStateAPIContext, RuntimeStateContext } from '@/contexts';
+import { useTheme } from '@/hooks';
+import { noop } from '@/utils/functional';
+import { useResolveAssistantStyleSheet } from '@/utils/stylesheet';
+
+import { ChatAPI } from '../ChatWidget/types';
+import ChatWindow from '.';
+
+interface ChatWindowStandaloneViewProps extends React.PropsWithChildren {
+  chatAPI?: ChatAPI | undefined;
+  ready?: () => void;
+}
+
+const ChatWindowStandaloneView: React.FC<ChatWindowStandaloneViewProps> = ({ chatAPI, ready }) => {
+  const { shadowRoot } = useContext(RuntimeStateContext);
+  const { assistant, interact } = useContext(RuntimeStateAPIContext);
+  const theme = useTheme(assistant);
+
+  const isStyleSheetResolved = useResolveAssistantStyleSheet(assistant, shadowRoot);
+
+  useEffect(() => {
+    if (!isObject(chatAPI)) return undefined;
+    console.info('Methods open, close, hide, show, proactive.clear, proactive.push have no effect in this mode.');
+
+    Object.assign(chatAPI, {
+      open: noop,
+      close: noop,
+      hide: noop,
+      show: noop,
```

Comment:
instead of purely `noop` we could console warn a warning to the user so know they they can't call this function during a certain mode?
Might help them with debugging more

Reply:
good idea 👍